sdk/monitor/monitor-opentelemetry/src/shared/logging/diagFileConsoleLogger.ts

```diff
@@ -169,6 +192,23 @@ export class DiagFileConsoleLogger implements DiagLogger {
     return false;
   }
 
+  private _shouldFilterAzureMonitorExporterWarning(message?: any): boolean {
+    if (typeof message !== "string") {
+      return false;
+    }
+
+    const text = message.toLowerCase();
+    const matchesUnsupportedValue =
+      text.includes("unsupported otel_metrics_exporter value") ||
+      (text.includes("otel_metrics_exporter value") && text.includes("supported values are"));
+
+    return (
+      text.includes("otel_metrics_exporter") &&
+      text.includes("azure_monitor") &&
+      matchesUnsupportedValue
```

Comment:
Move this `matchesUnsupportedValue` to the first argument so that at runtime this will shortcut the checks and avoid the 2 additional `text.includes` as this looks like it's going to run on every log message.

It look like you could also add a check for `otel_metrics_exporter` as the first thing (after the lowercase) to avoid several more `.includes`